### DISCUSSION TITLE
Is it possible to measure another element?

### DISCUSSION THREAD

**Discussion by Zloka** [thebuilder/react-intersection-observer#494] [Q&A]
Hi, thanks for a very convenient hook!

I was wondering if it is possible to observe _some other_ element? I.e. I have a component, but I would not like to assign the `ref` to the component I am returning, rather I want _this component_ to observe and take action based on whether or not another component is intersecting the viewport. I figured I would find it e.g. using `document.getElementById`.

Effectively, I am trying to detect whether or not a toolbar is active, as it has a `fixed` position at the top of the screen and is overlapping some of my content. Sadly, it's a piece of functionality integrated from a third party, so I don't have much control over it in my React tree/state management.

**Comment by thebuilder** [comment]
You can call the `ref()` with an element inside an `useEffect`. This would allow you set it to the element you get from `getElementById`.

**Comment by thebuilder** [comment]
See https://github.com/thebuilder/react-intersection-observer/issues/480 for a few different ideas
